src/Features/Core/Portable/ConvertNumericLiteral/AbstractConvertNumericLiteralCodeRefactoringProvider.cs

```diff
@@ -4,169 +4,134 @@
-            }
+        var numericToken = numericLiteralNode.GetFirstToken();
+        if (numericToken.ContainsDiagnostics)
+            return;
 
-            var syntaxNode = numericToken.GetRequiredParent();
-            var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
-            var symbol = semanticModel.GetTypeInfo(syntaxNode, cancellationToken).Type;
-            if (symbol == null)
-            {
-                return;
-            }
+        var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+        var symbol = semanticModel.GetTypeInfo(numericLiteralNode, cancellationToken).Type;
+        if (symbol == null)
+            return;
 
-            if (!IsIntegral(symbol.SpecialType))
-            {
-                return;
-            }
+        if (!symbol.SpecialType.IsIntegralType())
+            return;
 
-            var valueOpt = semanticModel.GetConstantValue(syntaxNode);
-            if (!valueOpt.HasValue)
-            {
-                return;
-            }
+        var valueOpt = semanticModel.GetConstantValue(numericLiteralNode);
+        if (!valueOpt.HasValue)
+            return;
 
-            var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+        var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
 
-            var value = IntegerUtilities.ToInt64(valueOpt.Value);
-            var numericText = numericToken.ToString();
-            var (hexPrefix, binaryPrefix) = GetNumericLiteralPrefixes();
-            var (prefix, number, suffix) = GetNumericLiteralParts(numericText, hexPrefix, binaryPrefix);
-            var kind = string.IsNullOrEmpty(prefix) ? NumericKind.Decimal
-                : prefix.Equals(hexPrefix, StringComparison.OrdinalIgnoreCase) ? NumericKind.Hexadecimal
-                : prefix.Equals(binaryPrefix, StringComparison.OrdinalIgnoreCase) ? NumericKind.Binary
-                : NumericKind.Unknown;
+        var value = IntegerUtilities.ToInt64(valueOpt.Value);
+        var numericText = numericToken.ToString();
+        var (prefix, number, suffix) = GetNumericLiteralParts(numericText);
+        var kind = string.IsNullOrEmpty(prefix) ? NumericKind.Decimal
+            : prefix.Equals(hexPrefix, StringComparison.OrdinalIgnoreCase) ? NumericKind.Hexadecimal
+            : prefix.Equals(binaryPrefix, StringComparison.OrdinalIgnoreCase) ? NumericKind.Binary
+            : NumericKind.Unknown;
 
-            if (kind == NumericKind.Unknown)
-            {
-                return;
-            }
+        if (kind == NumericKind.Unknown)
+            return;
 
-            if (kind != NumericKind.Decimal)
-            {
-                RegisterRefactoringWithResult(value.ToString(), FeaturesResources.Convert_to_decimal);
-            }
+        using var result = TemporaryArray<CodeAction>.Empty;
```

Comment:
build a list of actions that can be shown as nested actions if there are too many.